villainary-react/app/profile/Profile.tsx

```diff
@@ -45,19 +65,33 @@ export default function Profile() {
               formState,
             }) => (
               <TextField
-                {...register('villainName', { required: true })}
+                {...register('villainName', {
+                  required: 'This field is required',
+                })}
                 inputRef={ref}
-                onChange={onChange}
+                onChange={(value) => {
+                  onChange(value);
+                  changeName(value);
```

Comment:
What is happening here?  Why is onChange calling onChange?

Reply:
the first is the onchange binding of the text field, the second is the onChange method provided by react hook forms